python_modules/dagster/dagster/_core/definitions/asset_node.py

```diff
@@ -0,0 +1,47 @@
+from typing import Iterable, Optional, Union
+
+from .auto_materialize_policy import AutoMaterializePolicy
+from .events import (
+    AssetKey,
+    CoercibleToAssetKey,
+)
+from .freshness_policy import FreshnessPolicy
+from .metadata import MetadataUserInput
+
+
+class AssetNode:
```

Comment:
I think this should inherit from a class that enforces that this is a value. Are we not using NamedTuple anymore?